### ISSUE TITLE
[Bug?] Using `info replication` for replica discovery yields private ips

### ISSUE THREAD

**Issue by snapbug** [aembke/fred.rs#132]
Redis version - 6.0.5 via ElastiCache
Platform - n/a
Using Docker and/or Kubernetes - n/an
Deployment type - cluster

**Describe the bug**
When connecting to replicas fred parses the response of the `info replication` command to find the replicas for a given node in the cluster [[relevant pointer](https://github.com/aembke/fred.rs/blob/4668d41136938fc5dd42b4d2a8a71a789e7cb42e/src/router/replicas.rs#LL359C1-L359C1)].

For ElastiCache (and possibly others), when compared to the `cluster slots` command, this yields at best, a different IP address and at worse an address that mightn't be accessible.

**To Reproduce**
Problematically I think this might rely on an AWS setup thing, rather than generalizable outside.

Output of `cluster slots`:
```
...
xx) 1) (integer) 16242                               
    2) (integer) 16383                               
    3) 1) "10.210.aaa.bbb"                           
       2) (integer) 6379                             
       3) "1a13137169931ad8db388fae36e96b9fbb84b672" 
    4) 1) "10.210.ccc.ddd"                            
       2) (integer) 6379                             
       3) "a57886a0ea4123a2a63a32f4c907b6c90f84d06d" 
```

From the master side:
```
❯ timeout 3 redis-cli -h 10.210.aaa.bbb -p 6379 info replication
# Replication
role:master
connected_slaves:1
slave0:ip=10.31.yyy.zzz,port=6379,state=online,offset=5541916556922,lag=1
...
```

And from the replica side using the ip from `cluster slots`:
```
❯ timeout 3 redis-cli -h 10.210.ccc.ddd -p 6379 info replication
# Replication
role:slave
master_host:10.210.aaa.bbb
master_port:6379
...
```

... and using the ip from the `info replication` output:
```
❯ timeout 3 redis-cli -h 10.31.yyy.zzz -p 6379 info replication
❯ echo $?
124
```

**Logs**
(Set `RUST_LOG=fred=trace` and run with `--features debug-ids`)
Can provide.

**Additional context**

**Comment by aembke** [comment]
Oh that's interesting, thanks for pointing this out. I'll switch the replica logic to use `CLUSTER SLOTS` rather than `INFO replication` when possible then.

**Comment by aembke** [comment]
@snapbug this should be fixed in 6.3.0. The implementation now uses `ROLE` instead of `INFO replication`, but only when the replica info cannot be found in the `CLUSTER SLOTS` response first. Let me know if you still see issues here though.

**Comment by snapbug** [comment]
Looks good, thanks 🙂
